<a id='sommaire'></a>
<h1 style='text-align:center'>FAMILIARISATION AVEC LES DONN&Eacute;ES</h1>

[colonne id_prod](#colonne_id_prod)  
[colonne date](#colonne_date)  
[colonne session_id](#colonne_session_id)  
[colonne client_id](#colonne_client_id)  
[cas particuliers](#cas_particuliers)  
[table client](#table_client)  
[colonne sex](#colonne_sex)  
[colonne birth](#colonne_birth)  
[ajout de la colonne âge](#ajout_age)  
[ajout de la colonne tranche d'âge](#ajout_tranche_age)  
[table products](#table_products)  
[colonne id_prod](#colonne_id_prod)  
[colonne price](#colonne_price)  
[colonne categ](#colonne_categ)  
[ajout de la colonne tranche de prix](#ajout_tranche_prix)  
[création du fichier unique](#fichier_unique)  
[deuxième jointure ](#jointure_trans_cust_prod)  
[création du dataframe produits_non_vendus](#produits_non_vendus)  
[ajout des colones taille panier et achats](#deux_ajout)  
[étude des dataframe annexes](#df_annexes)  
[clients grands comptes](#grands_comptes)  
[ventes normales](#vente_normale)

## La table 'transactions'

<h3 style='color:#8677b2'>Lecture du CSV transactions</h3>

In [1]:
# -*- coding: utf-8 -*-

import pandas as pd
transactions = pd.read_csv("transactions.csv")

<h3 style='color:#8677b2'>Premières informations</h3>

In [2]:
a = transactions.groupby(['date', 'id_prod', 'session_id', 'client_id']).count()
a.shape, transactions.shape

((336890, 0), (337016, 4))

In [3]:
transactions.head(3)

id_prod                        date session_id client_id
0  0_1483  2021-04-10 18:37:28.723910    s_18746    c_4450
1   2_226  2022-02-03 01:55:53.276402   s_159142     c_277
2   1_374  2021-09-23 15:13:46.938559    s_94290    c_4270

In [4]:
transactions.shape

(337016, 4)

<p style='color:#88cc8d;text-align:center'>
    Quatre colonnes: id_prod, date, session_id et client_id<br/> 
    337 016 lignes<br/> 
</p>

<h3 style='color:#8677b2'>Travail sur la colonne id_prod</h3><a id='colonne_id_prod></a>

[Haut de page](#sommaire)

In [5]:
tra1 = transactions.groupby('id_prod').count()
tra1 = tra1.reset_index()
tra1 = tra1.sort_values(by='id_prod')
tra1.head()
tra1

id_prod  date  session_id  client_id
0        0_0   611         611        611
1        0_1   249         249        249
2       0_10    12          12         12
3      0_100     2           2          2
4     0_1000   222         222        222
5     0_1001   211         211        211
6     0_1002   231         231        231
7     0_1003   219         219        219
8     0_1004   206         206        206
9     0_1005     3           3          3
10    0_1006     4           4          4
11    0_1007   216         216        216
12    0_1008   196         196        196
13    0_1009   218         218        218
14     0_101     8           8          8
15    0_1010     7           7          7
16    0_1011   208         208        208
17    0_1012     2           2          2
18    0_1013   222         222        222
19    0_1015   231         231        231
20    0_1017   219         219        219
21    0_1018   215         215        215
22    0_1019     3           3          3
23     0_102     7           7          7
24    0_1020   218         218        218
25    0_1021   194         194        194
26    0_1022     4           4          4
27    0_1023   265         265        265
28    0_1024     2           2          2
29    0_1026   240         240        240
...      ...   ...         ...        ...
3236    2_70     8           8          8
3237    2_71     5           5          5
3238    2_73     6           6          6
3239    2_74     4           4          4
3240    2_75     2           2          2
3241    2_76     4           4          4
3242    2_77     1           1          1
3243    2_78     3           3          3
3244    2_79   189         189        189
3245     2_8     3           3          3
3246    2_80     5           5          5
3247    2_81     1           1          1
3248    2_82     4           4          4
3249    2_83     5           5          5
3250    2_84     5           5          5
3251    2_85     3           3          3
3252    2_88     3           3          3
3253    2_89     2           2          2
3254     2_9   180         180        180
3255    2_90     2           2          2
3256    2_91     4           4          4
3257    2_92     8           8          8
3258    2_93     1           1          1
3259    2_94     5           5          5
3260    2_95     3           3          3
3261    2_96   281         281        281
3262    2_97     5           5          5
3263    2_98     1           1          1
3264    2_99     2           2          2
3265     T_0   200         200        200

[3266 rows x 4 columns]

In [6]:
tra1.tail()

id_prod  date  session_id  client_id
3261    2_96   281         281        281
3262    2_97     5           5          5
3263    2_98     1           1          1
3264    2_99     2           2          2
3265     T_0   200         200        200

In [7]:
tra1.shape

(3266, 4)

<p style='color:#88cc8d;text-align:center'>
    3 266 références de produits ont été vendus<br/> 
</p>

<p style='color:#8677b2;text-align:center'>
    Les id_prod semble être de la forme Lettre_Numeros excepté T_0<br/>
</p>

In [8]:
tra2 = tra1.sort_values(by='date')
tra2.head()

id_prod  date  session_id  client_id
665   0_1601     1           1          1
3247    2_81     1           1          1
2634   1_408     1           1          1
166   0_1151     1           1          1
3242    2_77     1           1          1

In [9]:
tra2.tail()

id_prod  date  session_id  client_id
2653   1_425  1013        1013       1013
2641   1_414  1027        1027       1027
2733   1_498  1036        1036       1036
2644   1_417  1062        1062       1062
2591   1_369  1081        1081       1081

<p style='color:#88cc8d;text-align:center'>
    les 5 produits les plus vendus l'ont été environ 1 000 fois<br/> 
</p>

<a id='colonne_date'></a>
<h3 style='color:#8677b2'>Travail sur la colonne date</h3>

[haut de page](#sommaire)

In [10]:
tra3 = transactions.groupby('date').count()
tra3 = tra3.reset_index()
tra3 = tra3.sort_values(by='date')
tra3.head()

date  id_prod  session_id  client_id
0  2021-03-01 00:01:07.843138        1           1          1
1  2021-03-01 00:02:26.047414        1           1          1
2  2021-03-01 00:02:38.311413        1           1          1
3  2021-03-01 00:04:54.559692        1           1          1
4  2021-03-01 00:05:18.801198        1           1          1

In [11]:
tra3.tail()

date  id_prod  session_id  client_id
336850  test_2021-03-01 02:30:02.237446        7           7          7
336851  test_2021-03-01 02:30:02.237447        3           3          3
336852  test_2021-03-01 02:30:02.237448        1           1          1
336853  test_2021-03-01 02:30:02.237449        3           3          3
336854  test_2021-03-01 02:30:02.237450        1           1          1

In [12]:
tra3.shape

(336855, 4)

<p style='color:#88cc8d;text-align:center'>
    336 855 dates au format ISO 8601<br/> 
</p>

<p style='color:#8677b2;text-align:center'>
    ATTENTION: Il existe des dates qui commencent par test<br/>
</p>

In [13]:
tra4 = tra3.sort_values('id_prod')
tra4.head()

date  id_prod  session_id  client_id
0       2021-03-01 00:01:07.843138        1           1          1
224549  2021-11-08 19:50:53.082948        1           1          1
224548  2021-11-08 19:48:51.965818        1           1          1
224547  2021-11-08 19:46:59.642536        1           1          1
224546  2021-11-08 19:42:14.270127        1           1          1

In [14]:
tra4.tail()

date  id_prod  session_id  client_id
336818  test_2021-03-01 02:30:02.237414        8           8          8
336832  test_2021-03-01 02:30:02.237428        8           8          8
336847  test_2021-03-01 02:30:02.237443        8           8          8
336831  test_2021-03-01 02:30:02.237427        8           8          8
336817  test_2021-03-01 02:30:02.237413       13          13         13

<p style='color:#8677b2;text-align:center'>
    ATTENTION: Il existe des dates qui comportent plusieurs sessions<br/>
</p>

<a id='colonne_session_id'></a>
<h3 style='color:#8677b2'>Travail sur la colonne session_id</h3>

[Haut de page](#sommaire)

In [15]:
tra5 = transactions.groupby('session_id').count()
tra5 = tra5.reset_index()
tra5 = tra5.sort_values(by='session_id')
tra5.head()

session_id  id_prod  date  client_id
0        s_0      200   200        200
1        s_1        1     1          1
2       s_10        1     1          1
3      s_100        2     2          2
4     s_1000        4     4          4

In [16]:
tra5.tail()

session_id  id_prod  date  client_id
169190    s_99994        3     3          3
169191    s_99995        1     1          1
169192    s_99996        4     4          4
169193    s_99997        1     1          1
169194    s_99998        3     3          3

In [17]:
tra5.shape

(169195, 4)

<p style='color:#88cc8d;text-align:center'>
    session_id est de la forme s_Numero<br/>
    Il y a 169 195 sessions<br/>
</p>

In [18]:
tra6 = tra5.sort_values(by='id_prod')
tra6.head()

session_id  id_prod  date  client_id
75223    s_170300        1     1          1
86115     s_24922        1     1          1
157927    s_89554        1     1          1
86113     s_24920        1     1          1
86112      s_2492        1     1          1

In [19]:
tra6.tail(6)

session_id  id_prod  date  client_id
73288    s_168560       12    12         12
162747    s_93892       12    12         12
81764     s_21005       13    13         13
166041    s_96857       13    13         13
17852    s_118668       14    14         14
0             s_0      200   200        200

<p style='color:#88cc8d;text-align:center'>
    Une dizaine de références pour les plus importantes sessions
</p>

<p style='color:#8677b2;text-align:center'>
    ATTENTION: La session s_0 a un poids bien supérieur aux autres session_id<br/>
</p>

<a id='colonne_client_id'></a>
<h3 style='color:#8677b2'>Travail sur la colonne client_id</h3>

[Haut de page](#sommaire)

In [20]:
tra7 = transactions.groupby('client_id').count()
tra7 = tra7.reset_index()
tra7.sort_values(by='client_id')
tra7.head()

client_id  id_prod  date  session_id
0       c_1       20    20          20
1      c_10       28    28          28
2     c_100        6     6           6
3    c_1000       56    56          56
4    c_1001       58    58          58

In [21]:
tra7.tail()

client_id  id_prod  date  session_id
8597     c_997       23    23          23
8598     c_998       28    28          28
8599     c_999       22    22          22
8600      ct_0      106   106         106
8601      ct_1       94    94          94

In [22]:
tra7.shape

(8602, 4)

<p style='color:#88cc8d;text-align:center'>
    client_id est de la forme c_Numero<br/>
    Il y a 8 602 clients acheteurs<br/>
</p>

<p style='color:#8677b2;text-align:center'>
    ATTENTION: Les client_id ct_0 et ct_1 ne sont pas de la forme c_Numero<br/>
</p>

In [23]:
tra8 = tra7.sort_values(by='id_prod')
tra8.head()

client_id  id_prod  date  session_id
7471    c_7739        1     1           1
7768    c_8005        1     1           1
1363    c_2226        1     1           1
8480     c_890        1     1           1
4078    c_4679        1     1           1

In [24]:
tra8.tail()

client_id  id_prod   date  session_id
1268    c_2140      195    195         195
4388    c_4958     2562   2562        2562
2724    c_3454     3275   3275        3275
6337    c_6714     4473   4473        4473
677     c_1609    12855  12855       12855

<p style='color:#8677b2;text-align:center'>
    ATTENTION: 1 client a commandé plus de 10 000 produits, 3 clients ont commandé plusieurs milliers de produits, alors que le 5ème a commandé 195 produits<br/>
</p>

<a id='cas_particuliers'></a>
<h3 style='color:#8677b2'>Focus sur les cas particuliers relevés</h3>

[Haut de page](#sommaire)

In [25]:
tra9 = transactions[transactions['id_prod'] == 'T_0']
tra9 = tra9.groupby(['id_prod', 'client_id', 'session_id']).count()
tra9.head()

date
id_prod client_id session_id      
T_0     ct_0      s_0          106
        ct_1      s_0           94

<p style='color:#88cc8d;text-align:center'>
    
</p>

In [26]:
test = transactions['date'].tolist()
tra10 = transactions.copy()
tra10['test'] = [True if 'test' in a else False for a in test]
tra11 = tra10[tra10['test'] == True]
tra11 = tra11.groupby(['test', 'session_id', 'client_id', 'id_prod']).count()
tra11

date
test session_id client_id id_prod      
True s_0        ct_0      T_0       106
                ct_1      T_0        94

In [27]:
tra12 = transactions[transactions['session_id'] == 's_0']
tra12 = tra12.groupby(['session_id', 'id_prod', 'client_id']).count()
tra12

date
session_id id_prod client_id      
s_0        T_0     ct_0        106
                   ct_1         94

<p style='color:#88cc8d;text-align:center'>
    Remarque: Les 200 lignes du produit suspect T_0 correspondent aux 200 lignes de la session suspecte s_0, qui correspondent aux 200 lignes des clients supects ct_0 et ct_1, qui correspondent aux 200 lignes de dates tests.<br/>
    Il convient de retirer ces 200 lignes qui sont des tests.
</p>

<h3 style='color:#8677b2'>Retrait des 200 lignes tests</h3>

In [28]:
transactions['test'] = [True if 'test' in a else False for a in test]
clean_transactions = transactions[transactions['test'] == False].copy()
clean_transactions = clean_transactions.drop(columns='test')

<h3 style='color:#8677b2'>Création des colonnes jour_sem, nom_mois, heure et partie_jour<br/>
Date devient session_date<br/>
La colonne date devient la date au format YYYY MM DD</h3>

In [29]:
clean_transactions['session_date'] = pd.to_datetime(clean_transactions['date'])

clean_transactions['jour_sem'] = clean_transactions['session_date'].map(lambda d: d.day)
clean_transactions['nom_mois'] = clean_transactions['session_date'].map(lambda d: d.month)
clean_transactions['heure'] = clean_transactions['session_date'].map(lambda d: d.hour)

def partie_jour(value):
    if 14 <= value < 22:
        return 'Afternoon'
    elif 6 <= value < 14:
        return 'Morning'
    else:
        return 'Nigth'

clean_transactions['partie_jour'] = clean_transactions['heure'].map(partie_jour)

clean_transactions['date'] = clean_transactions['session_date'].dt.strftime('%Y %m %d')
clean_transactions.head()

id_prod        date session_id client_id               session_date  \
0  0_1483  2021 04 10    s_18746    c_4450 2021-04-10 18:37:28.723910   
1   2_226  2022 02 03   s_159142     c_277 2022-02-03 01:55:53.276402   
2   1_374  2021 09 23    s_94290    c_4270 2021-09-23 15:13:46.938559   
3  0_2186  2021 10 17   s_105936    c_4597 2021-10-17 03:27:18.783634   
4  0_1351  2021 07 17    s_63642    c_1242 2021-07-17 20:34:25.800563   

   jour_sem  nom_mois  heure partie_jour  
0        10         4     18   Afternoon  
1         3         2      1       Nigth  
2        23         9     15   Afternoon  
3        17        10      3       Nigth  
4        17         7     20   Afternoon

In [30]:
clean_transactions.to_csv('clean_transactions.csv', index=False)
clean_transactions.shape

(336816, 9)

<p style='color:#88cc8d;text-align:center'>
    clean_transaction contient 336 816 lignes<br/>
    9 colonnes, 5 que l'on a crées<br/>
</p>

<a id='table_client'></a>
## La table client

<h3 style='color:#8677b2'>Lecture du CSV customers</h3>

[Haut de page](#sommaire)

In [31]:
customers = pd.read_csv("customers.csv")

<h3 style='color:#8677b2'>Premières informations</h3>

In [32]:
customers.head()

client_id sex  birth
0    c_4410   f   1967
1    c_7839   f   1975
2    c_1699   f   1984
3    c_5961   f   1962
4    c_5320   m   1943

In [33]:
customers.shape

(8623, 3)

<p style='color:#88cc8d;text-align:center'>
    customers contient 3 colonnes: client_id, sex, birth et 8 623 lignes<br/>
</p>

<h3 style='color:#8677b2'>Travail sur la colonne client_id</h3>
<a id='colonne_client_id2>

[Haut de page](#sommaire)

In [34]:
cus1 = customers.groupby('client_id').count()
cus1 = cus1.reset_index()
cus1 = cus1.sort_values(by='client_id')
cus1.head()

client_id  sex  birth
0       c_1    1      1
1      c_10    1      1
2     c_100    1      1
3    c_1000    1      1
4    c_1001    1      1

In [35]:
cus1.tail()

client_id  sex  birth
8618     c_997    1      1
8619     c_998    1      1
8620     c_999    1      1
8621      ct_0    1      1
8622      ct_1    1      1

In [36]:
cus1.shape

(8623, 3)

<p style='color:#88cc8d;text-align:center'>
    Il n'y a pas d'identifiant en doublon<br/>
    client_id de customers et client_id de transactions correspondent<br/>
    Le fichier contient 8 623 clients
</p>

In [37]:
cus2 = cus1.sort_values(by='sex')
cus2.head()

client_id  sex  birth
0          c_1    1      1
5756     c_618    1      1
5755    c_6179    1      1
5754    c_6178    1      1
5753    c_6177    1      1

In [38]:
cus2.tail()

client_id  sex  birth
2867     c_358    1      1
2866    c_3579    1      1
2865    c_3578    1      1
2879    c_3590    1      1
8622      ct_1    1      1

<p style='color:#8677b2;text-align:center'>
    ATTENTION: customers contient ct_1
</p>

<a id='colonne_sex'></a>
<h3 style='color:#8677b2'>Travail sur la colonne sex</h3>

[Haut de page](#sommaire)

In [39]:
cus3 = customers.groupby('sex').count()
cus3 = cus3.reset_index()
cus3 = cus3.sort_values(by='sex')
cus3.head()

sex  client_id  birth
0   f       4491   4491
1   m       4132   4132

<p style='color:#88cc8d;text-align:center'>
    Il y a 4 491 femmes et 4 132 hommes<br/>
</p>

<a id='colonne_birth'></a>
<h3 style='color:#8677b2'>Travail sur la colonne birth</h3>

[Haut de page](#sommaire)

In [40]:
cus4 = customers.groupby('birth').count()
cus4 = cus4.reset_index()
cus4 = cus4.sort_values(by='birth')
cus4.head()

birth  client_id  sex
0   1929          3    3
1   1930          4    4
2   1931          4    4
3   1932          6    6
4   1933          8    8

In [41]:
cus4.tail()

birth  client_id  sex
71   2000        136  136
72   2001        129  129
73   2002        146  146
74   2003        146  146
75   2004        440  440

In [42]:
cus4.shape

(76, 3)

<p style='color:#88cc8d;text-align:center'>
    Les années s'étalent de 1929 (93 ans) à 2004 (18 ans)<br/>
</p>

<p style='color:#8677b2;text-align:center'>
    ATTENTION: L'année 2004 semble avoir un poids plus important que les années suivantes<br/>
</p>

<a id='ajout_age'></a>
<h3 style='color:#8677b2'>Ajout de la colonne âge</h3>

[Haut de page](#sommaire)

In [43]:
customers['age'] = customers['birth'].map(lambda x: (2022 - x))
customers.head()

client_id sex  birth  age
0    c_4410   f   1967   55
1    c_7839   f   1975   47
2    c_1699   f   1984   38
3    c_5961   f   1962   60
4    c_5320   m   1943   79

<a id='ajout_tranche_age'></a>
<h3 style='color:#8677b2'>Ajout de la colonne tranche d'âge et tranche d'âge1</h3>

[Haut de page](#sommaire)

In [44]:
borne_age = [17, 30, 50, 100]
tranche_age = [f"{borne_age[x]+1} - {borne_age[x+1]}" for x in range(len(borne_age) - 1)]
customers['tranche d\'age'] = pd.cut(customers['age'], borne_age, labels=tranche_age)

borne_age1 = [17, 30, 100]
tranche_age1 = [f"{borne_age1[x]+1} - {borne_age1[x+1]}" for x in range(len(borne_age1) - 1)]
customers['tranche d\'age1'] = pd.cut(customers['age'], borne_age1, labels=tranche_age1)

a = customers[(customers['age'] == 20) | (customers['age'] == 40) | (customers['age'] == 60)]
a.head()

client_id sex  birth  age tranche d'age tranche d'age1
3     c_5961   f   1962   60      51 - 100       31 - 100
10    c_1013   m   1982   40       31 - 50       31 - 100
25    c_1361   f   1982   40       31 - 50       31 - 100
38    c_3991   f   1982   40       31 - 50       31 - 100
41     c_125   m   2002   20       18 - 30        18 - 30

<h3 style='color:#8677b2'>Retrait des client_id 'ct_0' et 'ct_0'</h3>

In [45]:
clean_customers = customers[(customers['client_id'] != 'ct_0') & (customers['client_id'] != 'ct_1')]
clean_customers.shape

(8621, 6)

In [46]:
clean_customers.to_csv('clean_customers.csv', index=False)
clean_customers.shape

(8621, 6)

<p style='color:#88cc8d;text-align:center'>
    clean_customers contient 8 621 clients<br/>
    Les colonnes âge, tranche d'âge et tranche d'âge1 ont été ajoutées<br/>
</p>

<a id='table_products'></a>
## La table products

<h3 style='color:#8677b2'>Lecture du CSV products</h3>

[Haut de page](#sommaire)

In [47]:
products = pd.read_csv("products.csv")

<h3 style='color:#8677b2'>Premières informations</h3>

In [48]:
products.head()

id_prod  price  categ
0  0_1421  19.99      0
1  0_1368   5.13      0
2   0_731  17.99      0
3   1_587   4.99      1
4  0_1507   3.99      0

In [49]:
products.shape

(3287, 3)

<p style='color:#88cc8d;text-align:center'>
    Products contient trois colonnes id_prod, price et categ. Il y a 3 287 lignes<br/> 
</p>

<a id='colonne_id_prod'></a>
<h3 style='color:#8677b2'>Travail sur la colonne id_prod</h3>

[Haut de page](#sommaire)

In [50]:
pro1 = products.groupby('id_prod').count()
pro1 = pro1.reset_index()
pro1 = pro1.sort_values(by='id_prod')
pro1.head()

id_prod  price  categ
0     0_0      1      1
1     0_1      1      1
2    0_10      1      1
3   0_100      1      1
4  0_1000      1      1

In [51]:
pro1.tail()

id_prod  price  categ
3282    2_96      1      1
3283    2_97      1      1
3284    2_98      1      1
3285    2_99      1      1
3286     T_0      1      1

In [52]:
pro1.shape

(3287, 3)

<p style='color:#88cc8d;text-align:center'>
    Il n'y a pas d'identifiant en doublon<br/>
    id_prod de products et id_prod de transactions correspondent<br/>
    Le fichier contient 3 287 produits
</p>

In [53]:
pro2 = pro1.sort_values(by='price')
pro2.head()

id_prod  price  categ
0        0_0      1      1
2185   0_889      1      1
2186    0_89      1      1
2187   0_890      1      1
2188   0_891      1      1

In [54]:
pro2.tail()

id_prod  price  categ
1098  0_1987      1      1
1099  0_1988      1      1
1100  0_1989      1      1
1090   0_198      1      1
3286     T_0      1      1

<p style='color:#8677b2;text-align:center'>
    ATTENTION: products contient T_0
</p>

<a id='colonne_price'></a>
<h3 style='color:#8677b2'>Travail sur la colonne price</h3>

[Haut de page](#sommaire)

In [55]:
pro3 = products.groupby('price').count()
pro3 = pro3.reset_index()
pro3 = pro3.sort_values(by='price')
pro3.head()

price  id_prod  categ
0  -1.00        1      1
1   0.62        2      2
2   0.66        1      1
3   0.77        1      1
4   0.80        1      1

In [56]:
pro3.tail()

price  id_prod  categ
1450  233.54        1      1
1451  236.99        1      1
1452  247.22        1      1
1453  254.44        1      1
1454  300.00        1      1

In [57]:
pro3.shape

(1455, 3)

<p style='color:#88cc8d;text-align:center'>
    Les prix s'étalent de -1.00 à 300.00<br/>
    Il y a un éventail de 1 455 prix différents<br/>
</p>

In [58]:
pro4 = pro3.sort_values(by='id_prod')
pro4.head()

price  id_prod  categ
0    -1.00        1      1
929  21.29        1      1
928  21.23        1      1
927  21.22        1      1
926  21.18        1      1

In [59]:
pro4.tail()

price  id_prod  categ
405   8.99       74     74
258   5.99       78     78
147   3.99       80     80
307   6.99       84     84
208   4.99       85     85

<p style='color:#88cc8d;text-align:center'>
    Les prix les plus répendus sont au nombre d'environ 80 individus et se situent entre 3.99€ et 8.99€<br/>
</p>

<a id='colonne_categ'></a>
<h3 style='color:#8677b2'>Travail sur la colonne categ</h3>

[Haut de page](#sommaire)

In [60]:
pro5 = products.groupby('categ').count()
pro5 = pro5.reset_index()
pro5 = pro5.sort_values(by='categ')
pro5.head()

categ  id_prod  price
0      0     2309   2309
1      1      739    739
2      2      239    239

<p style='color:#88cc8d;text-align:center'>
    Il y a 3 catégories: 0, 1 et 2<br/>
    2 309 individus pour la categ 0, 739 pour la categ 1 et 239 pour la categ 2<br/>
</p>

In [61]:
pro6 = products[products['price'] <= 0]
pro6

id_prod  price  categ
731     T_0   -1.0      0

<p style='color:#88cc8d;text-align:center'>
    T_0 est le produit qui correspond au prix suspect -1.00<br/>
</p>

In [62]:
clean_products = products[products['price'] > 0].copy()
clean_products.shape

(3286, 3)

<p style='color:#88cc8d;text-align:center'>
    products contient 3 286 produits<br/>
</p>

<a id='ajout_tranche_prix'></a>
<h3 style='color:#8677b2'>Ajout de la colonne tranche de prix</h3>

[Haut de page](#sommaire)

In [63]:
borne_prix = [0, 2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 25, 30, 100, 301]

tranche_prix = [f'{borne_prix[x]+0.01} - {borne_prix[x+1]}€' for x in range(len(borne_prix) - 1)]

clean_products['tranche de prix'] = pd.cut(clean_products['price'], borne_prix, labels=tranche_prix)
clean_products.head()

clean_products.head()

id_prod  price  categ tranche de prix
0  0_1421  19.99      0     18.01 - 20€
1  0_1368   5.13      0       4.01 - 6€
2   0_731  17.99      0     16.01 - 18€
3   1_587   4.99      1       4.01 - 6€
4  0_1507   3.99      0       2.01 - 4€

In [64]:
clean_products.to_csv('clean_products.csv', index=False)
clean_products.shape

(3286, 4)

<a id='fichier_unique'></a>
## Création d'un fichier unique

<h3 style='color:#8677b2'>Recherche des valeurs nulles sur la jointure de clean_transactions et clean_customers</h3>

[Haut de page](#sommaire)

In [65]:
trans_cust = clean_transactions.merge(clean_customers, left_on='client_id', right_on='client_id', how='outer')
trans_cust.isnull().sum()

id_prod           21
date              21
session_id        21
client_id          0
session_date      21
jour_sem          21
nom_mois          21
heure             21
partie_jour       21
sex                0
birth              0
age                0
tranche d'age      0
tranche d'age1     0
dtype: int64

<h3 style='color:#8677b2'>Repérage des valeurs nulles</h3>

In [66]:
trans_cust['client non acheteur'] = pd.isnull(trans_cust['session_id'])
trans_cust.head()

id_prod        date session_id client_id               session_date  \
0  0_1483  2021 04 10    s_18746    c_4450 2021-04-10 18:37:28.723910   
1   1_596  2021 09 12    s_88567    c_4450 2021-09-12 02:11:24.774608   
2   1_278  2021 09 10    s_87835    c_4450 2021-09-10 15:09:01.555889   
3  0_1447  2021 05 25    s_39094    c_4450 2021-05-25 00:17:56.841054   
4  0_1357  2021 08 01    s_69919    c_4450 2021-08-01 10:43:25.039433   

   jour_sem  nom_mois  heure partie_jour sex  birth  age tranche d'age  \
0      10.0       4.0   18.0   Afternoon   f   1977   45       31 - 50   
1      12.0       9.0    2.0       Nigth   f   1977   45       31 - 50   
2      10.0       9.0   15.0   Afternoon   f   1977   45       31 - 50   
3      25.0       5.0    0.0       Nigth   f   1977   45       31 - 50   
4       1.0       8.0   10.0     Morning   f   1977   45       31 - 50   

  tranche d'age1  client non acheteur  
0       31 - 100                False  
1       31 - 100                False  
2       31 - 100                False  
3       31 - 100                False  
4       31 - 100                False

<h3 style='color:#8677b2'>Création d'un dataframe des clients n'ayant pas effectués d'achat</h3>

In [67]:
client_sans_achat = trans_cust[trans_cust['client non acheteur'] == True]
client_sans_achat = client_sans_achat[['client_id', 'sex', 'birth', 'age', 'tranche d\'age']]
client_sans_achat.shape

(21, 5)

In [68]:
client_sans_achat.to_csv('client_sans_achat.csv', index_label=False)

<h3 style='color:#8677b2'>Suppression des clients n'ayant pas fait d'achat</h3>

In [69]:
clean_trans_cust = trans_cust[trans_cust['client non acheteur'] == False]
clean_trans_cust = clean_trans_cust.drop(columns='client non acheteur')
clean_trans_cust.isnull().sum()

id_prod           0
date              0
session_id        0
client_id         0
session_date      0
jour_sem          0
nom_mois          0
heure             0
partie_jour       0
sex               0
birth             0
age               0
tranche d'age     0
tranche d'age1    0
dtype: int64

<a id='jointure_trans_cust_prod'></a>
<h3 style='color:#8677b2'>Recherche des valeurs nulles sur la jointure de clean_trans_cust et clean_products</h3>

[Haut de page](#sommaire)

In [70]:
trans_prod = clean_trans_cust.merge(clean_products, left_on='id_prod', right_on='id_prod', how='outer')
trans_prod.isnull().sum()

id_prod              0
date                22
session_id          22
client_id           22
session_date        22
jour_sem            22
nom_mois            22
heure               22
partie_jour         22
sex                 22
birth               22
age                 22
tranche d'age       22
tranche d'age1      22
price              103
categ              103
tranche de prix    103
dtype: int64

<h3 style='color:#8677b2'>Repérage des valeurs nulles (produit non vendu)</h3>

In [71]:
trans_prod['produit non vendu'] = pd.isnull(trans_prod['session_id'])

<a id='produits_non_vendus'></a>
<h3 style='color:#8677b2'>Création du dataframe des produits non vendus</h3>

[Haut de page](#sommaire)

In [72]:
produit_non_vendu = trans_prod[trans_prod['produit non vendu'] == True]
produit_non_vendu = produit_non_vendu[['id_prod', 'categ', 'price', 'tranche de prix']]
produit_non_vendu.shape

(22, 4)

In [73]:
produit_non_vendu.to_csv('produit_non_vendu.csv', index_label=False)

<h3 style='color:#8677b2'>Repérage des valeurs nulles (vente produit non référencé)</h3>

In [74]:
trans_prod['vente produit non référencé'] = pd.isnull(trans_prod['price'])

<h3 style='color:#8677b2'>Création du dataframe des vente de produits non référencés</h3>

In [75]:
produit_non_ref = trans_prod[trans_prod['vente produit non référencé'] == True]
produit_non_ref.shape

(103, 19)

In [76]:
produit_non_ref = produit_non_ref.drop(columns=(['produit non vendu', 'vente produit non référencé']))
produit_non_ref.groupby('id_prod').describe()

jour_sem                                                  nom_mois  \
           count       mean       std  min  25%   50%   75%   max    count   
id_prod                                                                      
0_2245     103.0  15.281553  8.634651  1.0  8.5  16.0  22.5  31.0    103.0   

                   ... price     categ                               
             mean  ...   75% max count mean std min 25% 50% 75% max  
id_prod            ...                                               
0_2245   6.446602  ...   NaN NaN   0.0  NaN NaN NaN NaN NaN NaN NaN  

[1 rows x 56 columns]

In [77]:
produit_non_ref.to_csv('produit_non_ref.csv', index_label=False)

<h3 style='color:#8677b2'>Création du dataframe des ventes sans le produit non référencé et sans les produits non vendus</h3>
<a id='vente'></a>

In [78]:
vente = trans_prod[(trans_prod['vente produit non référencé'] == False) &
                   (trans_prod['produit non vendu'] == False)]

vente = vente.drop(columns=['vente produit non référencé', 'produit non vendu'])
vente.isnull().sum()
vente.shape

(336713, 17)

<a id='deux_ajout'></a>
<h3 style='color:#8677b2'>Ajout de la colonne taille panier</h3>

[Haut de page](#sommaire)

In [79]:
panier = vente.groupby(['session_id'], as_index=False).count()

panier = panier[['session_id', 'id_prod']]

panier.rename(columns={'session_id':'session_id', 'id_prod':'taille_panier_session'}, inplace=True)

vente1 = vente.merge(panier, on='session_id')
vente1.head()

id_prod        date session_id client_id               session_date  \
0  0_1483  2021 04 10    s_18746    c_4450 2021-04-10 18:37:28.723910   
1   1_685  2021 04 10    s_18746    c_4450 2021-04-10 18:25:09.319011   
2  0_1395  2021 04 10    s_18746    c_4450 2021-04-10 18:32:27.342375   
3  0_1483  2021 12 27   s_140787    c_5433 2021-12-27 11:11:12.123067   
4  0_1483  2021 10 27   s_110736     c_857 2021-10-27 04:56:38.293970   

   jour_sem  nom_mois  heure partie_jour sex   birth   age tranche d'age  \
0      10.0       4.0   18.0   Afternoon   f  1977.0  45.0       31 - 50   
1      10.0       4.0   18.0   Afternoon   f  1977.0  45.0       31 - 50   
2      10.0       4.0   18.0   Afternoon   f  1977.0  45.0       31 - 50   
3      27.0      12.0   11.0     Morning   f  1981.0  41.0       31 - 50   
4      27.0      10.0    4.0       Nigth   m  1985.0  37.0       31 - 50   

  tranche d'age1  price  categ tranche de prix  taille_panier_session  
0       31 - 100   4.99    0.0       4.01 - 6€                      3  
1       31 - 100  15.99    1.0     14.01 - 16€                      3  
2       31 - 100   3.99    0.0       2.01 - 4€                      3  
3       31 - 100   4.99    0.0       4.01 - 6€                      1  
4       31 - 100   4.99    0.0       4.01 - 6€                      1

<h3 style='color:#8677b2'>Ajout des colonne achat_mensuel, achat_annuel et freq_mens</h3>

In [80]:
vente2 = vente.groupby(['client_id', 'nom_mois']).count()

achat_mensuel = vente2[['session_id']]

achat_annuel = vente['client_id'].value_counts()

achat = achat_mensuel.join(achat_annuel, on='client_id')

achat = achat.rename({'client_id':'client_id', 'nom_mois':'nom_mois',
                              'session_id':'achat_mens', 'client_id':'achat_an'},
                  axis='columns')

achat['freq_mens'] = achat['achat_mens'] / achat['achat_an']

achat = achat.reset_index()
vente3 = vente1.merge(achat, on=['client_id', 'nom_mois'])
vente3.head()

id_prod        date session_id client_id               session_date  \
0  0_1483  2021 04 10    s_18746    c_4450 2021-04-10 18:37:28.723910   
1   1_685  2021 04 10    s_18746    c_4450 2021-04-10 18:25:09.319011   
2  0_1395  2021 04 10    s_18746    c_4450 2021-04-10 18:32:27.342375   
3  0_1412  2021 04 09    s_18121    c_4450 2021-04-09 10:25:24.957469   
4  0_1483  2021 12 27   s_140787    c_5433 2021-12-27 11:11:12.123067   

   jour_sem  nom_mois  heure partie_jour sex  ...   age  tranche d'age  \
0      10.0       4.0   18.0   Afternoon   f  ...  45.0        31 - 50   
1      10.0       4.0   18.0   Afternoon   f  ...  45.0        31 - 50   
2      10.0       4.0   18.0   Afternoon   f  ...  45.0        31 - 50   
3       9.0       4.0   10.0     Morning   f  ...  45.0        31 - 50   
4      27.0      12.0   11.0     Morning   f  ...  41.0        31 - 50   

  tranche d'age1  price  categ  tranche de prix taille_panier_session  \
0       31 - 100   4.99    0.0        4.01 - 6€                     3   
1       31 - 100  15.99    1.0      14.01 - 16€                     3   
2       31 - 100   3.99    0.0        2.01 - 4€                     3   
3       31 - 100  11.73    0.0      10.01 - 12€                     1   
4       31 - 100   4.99    0.0        4.01 - 6€                     1   

   achat_mens  achat_an  freq_mens  
0           4       100   0.040000  
1           4       100   0.040000  
2           4       100   0.040000  
3           4       100   0.040000  
4          13        94   0.138298  

[5 rows x 21 columns]

In [81]:
vente3.to_csv('vente_totale.csv', index_label=False)

In [82]:
vente3.isnull().sum()

id_prod                  0
date                     0
session_id               0
client_id                0
session_date             0
jour_sem                 0
nom_mois                 0
heure                    0
partie_jour              0
sex                      0
birth                    0
age                      0
tranche d'age            0
tranche d'age1           0
price                    0
categ                    0
tranche de prix          0
taille_panier_session    0
achat_mens               0
achat_an                 0
freq_mens                0
dtype: int64

<h3 style='color:#8677b2'>Période de l'étude</h3>

In [83]:
vente3.sort_values('date').head(1)

id_prod        date session_id client_id               session_date  \
90640  0_1348  2021 03 01      s_120    c_1857 2021-03-01 05:28:38.663886   

       jour_sem  nom_mois  heure partie_jour sex  ...   age  tranche d'age  \
90640       1.0       3.0    5.0       Nigth   f  ...  32.0        31 - 50   

      tranche d'age1  price  categ  tranche de prix taille_panier_session  \
90640       31 - 100  12.03    0.0      12.01 - 14€                     3   

       achat_mens  achat_an  freq_mens  
90640          20       100        0.2  

[1 rows x 21 columns]

In [84]:
vente3.sort_values('date').tail(1)

id_prod        date session_id client_id               session_date  \
99664   1_468  2022 02 28   s_172339    c_1890 2022-02-28 19:53:05.366278   

       jour_sem  nom_mois  heure partie_jour sex  ...   age  tranche d'age  \
99664      28.0       2.0   19.0   Afternoon   f  ...  58.0       51 - 100   

      tranche d'age1  price  categ  tranche de prix taille_panier_session  \
99664       31 - 100  17.75    1.0      16.01 - 18€                     1   

       achat_mens  achat_an  freq_mens  
99664           5        61   0.081967  

[1 rows x 21 columns]

<p style='color:#88cc8d;text-align:center'>
    Le fichier vente démarre le 01-03-2021 et se termine le 28-02-2022<br/> 
</p>

In [85]:
vente3.groupby('session_date').count().shape

(336713, 20)

In [86]:
vente3.groupby('session_id').count().shape

(169173, 20)

In [87]:
vente3.groupby('client_id').count().shape

(8600, 20)

In [88]:
vente3.groupby('id_prod').count().shape

(3264, 20)

<p style='color:#88cc8d;text-align:center'>
    336 713 sessions_date<br/>
    169 173 sessions<br/> 
    8 600 clients acheteurs<br/> 
    3 264 références produits vendus<br/> 
</p>

<a id='df_annexes'></a>
## &Eacute;tude des dataframes annexes

[Haut de page](#sommaire)

<h3 style='color:#8677b2'>produit_non_ref</h3>

In [89]:
produit_non_ref.describe(include='all')

id_prod        date session_id client_id                session_date  \
count      103         103        103       103                         103   
unique       1          90        103       100                         103   
top     0_2245  2021 04 10    s_95930    c_1025  2021-09-25 17:07:45.271509   
freq       103           4          1         2                           1   
first      NaN         NaN        NaN       NaN  2021-03-01 00:09:29.301897   
last       NaN         NaN        NaN       NaN  2022-02-28 18:08:49.875709   
mean       NaN         NaN        NaN       NaN                         NaN   
std        NaN         NaN        NaN       NaN                         NaN   
min        NaN         NaN        NaN       NaN                         NaN   
25%        NaN         NaN        NaN       NaN                         NaN   
50%        NaN         NaN        NaN       NaN                         NaN   
75%        NaN         NaN        NaN       NaN                         NaN   
max        NaN         NaN        NaN       NaN                         NaN   

          jour_sem    nom_mois       heure partie_jour  sex        birth  \
count   103.000000  103.000000  103.000000         103  103   103.000000   
unique         NaN         NaN         NaN           3    2          NaN   
top            NaN         NaN         NaN       Nigth    f          NaN   
freq           NaN         NaN         NaN          35   60          NaN   
first          NaN         NaN         NaN         NaN  NaN          NaN   
last           NaN         NaN         NaN         NaN  NaN          NaN   
mean     15.281553    6.446602   11.009709         NaN  NaN  1977.262136   
std       8.634651    3.411998    6.833446         NaN  NaN    11.780643   
min       1.000000    1.000000    0.000000         NaN  NaN  1933.000000   
25%       8.500000    4.000000    4.000000         NaN  NaN  1973.000000   
50%      16.000000    6.000000   10.000000         NaN  NaN  1979.000000   
75%      22.500000    9.000000   17.000000         NaN  NaN  1984.000000   
max      31.000000   12.000000   23.000000         NaN  NaN  2002.000000   

               age tranche d'age tranche d'age1  price  categ  tranche de prix  
count   103.000000           103            103    0.0    0.0              0.0  
unique         NaN             3              2    NaN    NaN              0.0  
top            NaN       31 - 50       31 - 100    NaN    NaN              NaN  
freq           NaN            81            100    NaN    NaN              NaN  
first          NaN           NaN            NaN    NaN    NaN              NaN  
last           NaN           NaN            NaN    NaN    NaN              NaN  
mean     44.737864           NaN            NaN    NaN    NaN              NaN  
std      11.780643           NaN            NaN    NaN    NaN              NaN  
min      20.000000           NaN            NaN    NaN    NaN              NaN  
25%      38.000000           NaN            NaN    NaN    NaN              NaN  
50%      43.000000           NaN            NaN    NaN    NaN              NaN  
75%      49.000000           NaN            NaN    NaN    NaN              NaN  
max      89.000000           NaN            NaN    NaN    NaN              NaN

<p style='color:#88cc8d;text-align:center'>
    Un seul produit non référencé, vendu 103 fois sur toute l'année à 90 dates différentes et à 100 clients différents, sans distinction significative d'âge ou de sexe
</p>

<h3 style='color:#8677b2'>produit_non_vendu</h3>

In [90]:
produit_non_vendu.describe(include='all')

id_prod      categ       price tranche de prix
count       22  22.000000   22.000000              22
unique      22        NaN         NaN               5
top        1_0        NaN         NaN     20.01 - 25€
freq         1        NaN         NaN               8
mean       NaN   0.363636   36.285000             NaN
std        NaN   0.726731   55.753603             NaN
min        NaN   0.000000    0.800000             NaN
25%        NaN   0.000000    2.990000             NaN
50%        NaN   0.000000   21.485000             NaN
75%        NaN   0.000000   30.112500             NaN
max        NaN   2.000000  220.990000             NaN

In [91]:
produit_non_vendu.groupby('categ').describe()

price                                                                 
      count     mean        std     min       25%      50%       75%     max
categ                                                                       
0.0    17.0   13.650  11.611998    0.80    2.9900   20.080   22.9900   35.06
1.0     2.0   35.775   5.593215   31.82   33.7975   35.775   37.7525   39.73
2.0     3.0  164.890  48.790141  132.36  136.8400  141.320  181.1550  220.99

<p style='color:#88cc8d;text-align:center'>
    22 produits de 0.80€ à 220.99€, des trois catégories<br/> 
</p>

<h3 style='color:#8677b2'>client_sans_achat</h3>

In [92]:
client_sans_achat.describe(include='all')

client_id  sex        birth        age tranche d'age
count         21   21    21.000000  21.000000            21
unique        21    2          NaN        NaN             2
top         c_90    f          NaN        NaN       18 - 30
freq           1   11          NaN        NaN            13
mean         NaN  NaN  1984.380952  37.619048           NaN
std          NaN  NaN    19.878320  19.878320           NaN
min          NaN  NaN  1956.000000  18.000000           NaN
25%          NaN  NaN  1963.000000  21.000000           NaN
50%          NaN  NaN  1993.000000  29.000000           NaN
75%          NaN  NaN  2001.000000  59.000000           NaN
max          NaN  NaN  2004.000000  66.000000           NaN

<p style='color:#88cc8d;text-align:center'>
    21 clients de 18 à 66 ans, des deux sexes<br/> 
</p>

<a id='grands_comptes'></a>
## Clients grands comptes

[Haut de page](#sommaire)

In [93]:
vente3.head()

id_prod        date session_id client_id               session_date  \
0  0_1483  2021 04 10    s_18746    c_4450 2021-04-10 18:37:28.723910   
1   1_685  2021 04 10    s_18746    c_4450 2021-04-10 18:25:09.319011   
2  0_1395  2021 04 10    s_18746    c_4450 2021-04-10 18:32:27.342375   
3  0_1412  2021 04 09    s_18121    c_4450 2021-04-09 10:25:24.957469   
4  0_1483  2021 12 27   s_140787    c_5433 2021-12-27 11:11:12.123067   

   jour_sem  nom_mois  heure partie_jour sex  ...   age  tranche d'age  \
0      10.0       4.0   18.0   Afternoon   f  ...  45.0        31 - 50   
1      10.0       4.0   18.0   Afternoon   f  ...  45.0        31 - 50   
2      10.0       4.0   18.0   Afternoon   f  ...  45.0        31 - 50   
3       9.0       4.0   10.0     Morning   f  ...  45.0        31 - 50   
4      27.0      12.0   11.0     Morning   f  ...  41.0        31 - 50   

  tranche d'age1  price  categ  tranche de prix taille_panier_session  \
0       31 - 100   4.99    0.0        4.01 - 6€                     3   
1       31 - 100  15.99    1.0      14.01 - 16€                     3   
2       31 - 100   3.99    0.0        2.01 - 4€                     3   
3       31 - 100  11.73    0.0      10.01 - 12€                     1   
4       31 - 100   4.99    0.0        4.01 - 6€                     1   

   achat_mens  achat_an  freq_mens  
0           4       100   0.040000  
1           4       100   0.040000  
2           4       100   0.040000  
3           4       100   0.040000  
4          13        94   0.138298  

[5 rows x 21 columns]

In [94]:
gc = vente3.groupby('client_id').count()
gc.sort_values(by='id_prod', inplace=True)
gc.tail()

id_prod   date  session_id  session_date  jour_sem  nom_mois  \
client_id                                                                 
c_7959         195    195         195           195       195       195   
c_4958        2562   2562        2562          2562      2562      2562   
c_3454        3273   3273        3273          3273      3273      3273   
c_6714        4471   4471        4471          4471      4471      4471   
c_1609       12855  12855       12855         12855     12855     12855   

           heure  partie_jour    sex  birth    age  tranche d'age  \
client_id                                                           
c_7959       195          195    195    195    195            195   
c_4958      2562         2562   2562   2562   2562           2562   
c_3454      3273         3273   3273   3273   3273           3273   
c_6714      4471         4471   4471   4471   4471           4471   
c_1609     12855        12855  12855  12855  12855          12855   

           tranche d'age1  price  categ  tranche de prix  \
client_id                                                  
c_7959                195    195    195              195   
c_4958               2562   2562   2562             2562   
c_3454               3273   3273   3273             3273   
c_6714               4471   4471   4471             4471   
c_1609              12855  12855  12855            12855   

           taille_panier_session  achat_mens  achat_an  freq_mens  
client_id                                                          
c_7959                       195         195       195        195  
c_4958                      2562        2562      2562       2562  
c_3454                      3273        3273      3273       3273  
c_6714                      4471        4471      4471       4471  
c_1609                     12855       12855     12855      12855

<h3 style='color:#8677b2'>Création du fichier grands comptes</h3>

In [95]:
grands_comptes = vente3[(vente3['client_id'] == 'c_1609') |
      (vente3['client_id'] == 'c_6714') |
      (vente3['client_id'] == 'c_3454') |
      (vente3['client_id'] == 'c_4958')]

grands_comptes.to_csv('grands_comptes.csv', index_label=False)

<h3 style='color:#8677b2'>Création du fichier sans grands comptes</h3>

In [96]:
vente_normale = vente3[(vente3['client_id'] != 'c_1609') &
      (vente3['client_id'] != 'c_6714') &
      (vente3['client_id'] != 'c_3454') &
      (vente3['client_id'] != 'c_4958')]

vente_normale.to_csv('vente.csv', index_label=False)

In [97]:
print(100 * vente_normale['price'].sum()/vente3['price'].sum(),
100 * grands_comptes['price'].sum()/vente3['price'].sum())

92.51446582420303 7.485534175796971


<p style='color:#88cc8d;text-align:center'>
    Les 4 grands comptes représentent 7.5% du CA<br/> 
</p>

In [98]:
print(100 * vente_normale['id_prod'].count()/vente3['id_prod'].count(),
100 * grands_comptes['id_prod'].count()/vente3['id_prod'].count())

93.12144170257757 6.878558297422434


<p style='color:#88cc8d;text-align:center'>
    Les 4 grands comptes représentent 6.9% du volume des ventes<br/> 
</p>

<p style='color:#8677b2;text-align:center'>
    Pour notre étude, nous ne retiendront pas les clients grands comptes<br/>
</p>

<a id='vente_normale'></a>
<h3 style='color:#8677b2'>Ventes normales</h3>

[Haut de page](#sommaire)

In [99]:
nb_client = vente_normale.groupby('client_id').count()
nb_client.sort_values(by='id_prod', inplace=True)
nb_client.tail(10)

id_prod  date  session_id  session_date  jour_sem  nom_mois  heure  \
client_id                                                                       
c_5922         187   187         187           187       187       187    187   
c_2077         188   188         188           188       188       188    188   
c_3263         188   188         188           188       188       188    188   
c_7421         189   189         189           189       189       189    189   
c_8392         189   189         189           189       189       189    189   
c_3725         190   190         190           190       190       190    190   
c_8026         192   192         192           192       192       192    192   
c_2595         193   193         193           193       193       193    193   
c_2140         195   195         195           195       195       195    195   
c_7959         195   195         195           195       195       195    195   

           partie_jour  sex  birth  age  tranche d'age  tranche d'age1  price  \
client_id                                                                       
c_5922             187  187    187  187            187             187    187   
c_2077             188  188    188  188            188             188    188   
c_3263             188  188    188  188            188             188    188   
c_7421             189  189    189  189            189             189    189   
c_8392             189  189    189  189            189             189    189   
c_3725             190  190    190  190            190             190    190   
c_8026             192  192    192  192            192             192    192   
c_2595             193  193    193  193            193             193    193   
c_2140             195  195    195  195            195             195    195   
c_7959             195  195    195  195            195             195    195   

           categ  tranche de prix  taille_panier_session  achat_mens  \
client_id                                                              
c_5922       187              187                    187         187   
c_2077       188              188                    188         188   
c_3263       188              188                    188         188   
c_7421       189              189                    189         189   
c_8392       189              189                    189         189   
c_3725       190              190                    190         190   
c_8026       192              192                    192         192   
c_2595       193              193                    193         193   
c_2140       195              195                    195         195   
c_7959       195              195                    195         195   

           achat_an  freq_mens  
client_id                       
c_5922          187        187  
c_2077          188        188  
c_3263          188        188  
c_7421          189        189  
c_8392          189        189  
c_3725          190        190  
c_8026          192        192  
c_2595          193        193  
c_2140          195        195  
c_7959          195        195

In [100]:
nb_prod = vente_normale.groupby('id_prod').count()
nb_prod.sort_values(by='session_id', inplace=True)
nb_prod.tail(10)

date  session_id  client_id  session_date  jour_sem  nom_mois  heure  \
id_prod                                                                         
1_407     878         878        878           878       878       878    878   
1_406     881         881        881           881       881       881    881   
1_412     884         884        884           884       884       884    884   
1_398     891         891        891           891       891       891    891   
1_413     896         896        896           896       896       896    896   
1_425     941         941        941           941       941       941    941   
1_498     955         955        955           955       955       955    955   
1_414     966         966        966           966       966       966    966   
1_417     996         996        996           996       996       996    996   
1_369    1015        1015       1015          1015      1015      1015   1015   

         partie_jour   sex  birth   age  tranche d'age  tranche d'age1  price  \
id_prod                                                                         
1_407            878   878    878   878            878             878    878   
1_406            881   881    881   881            881             881    881   
1_412            884   884    884   884            884             884    884   
1_398            891   891    891   891            891             891    891   
1_413            896   896    896   896            896             896    896   
1_425            941   941    941   941            941             941    941   
1_498            955   955    955   955            955             955    955   
1_414            966   966    966   966            966             966    966   
1_417            996   996    996   996            996             996    996   
1_369           1015  1015   1015  1015           1015            1015   1015   

         categ  tranche de prix  taille_panier_session  achat_mens  achat_an  \
id_prod                                                                        
1_407      878              878                    878         878       878   
1_406      881              881                    881         881       881   
1_412      884              884                    884         884       884   
1_398      891              891                    891         891       891   
1_413      896              896                    896         896       896   
1_425      941              941                    941         941       941   
1_498      955              955                    955         955       955   
1_414      966              966                    966         966       966   
1_417      996              996                    996         996       996   
1_369     1015             1015                   1015        1015      1015   

         freq_mens  
id_prod             
1_407          878  
1_406          881  
1_412          884  
1_398          891  
1_413          896  
1_425          941  
1_498          955  
1_414          966  
1_417          996  
1_369         1015

In [101]:
nb_session = vente_normale.groupby('session_id').count()
nb_session.sort_values(by='session_date', inplace=True)
nb_session.tail(10)

id_prod  date  client_id  session_date  jour_sem  nom_mois  heure  \
session_id                                                                      
s_136877         11    11         11            11        11        11     11   
s_137262         11    11         11            11        11        11     11   
s_21641          11    11         11            11        11        11     11   
s_65873          11    11         11            11        11        11     11   
s_86536          11    11         11            11        11        11     11   
s_93892          12    12         12            12        12        12     12   
s_168560         12    12         12            12        12        12     12   
s_21005          13    13         13            13        13        13     13   
s_96857          13    13         13            13        13        13     13   
s_118668         14    14         14            14        14        14     14   

            partie_jour  sex  birth  age  tranche d'age  tranche d'age1  \
session_id                                                                
s_136877             11   11     11   11             11              11   
s_137262             11   11     11   11             11              11   
s_21641              11   11     11   11             11              11   
s_65873              11   11     11   11             11              11   
s_86536              11   11     11   11             11              11   
s_93892              12   12     12   12             12              12   
s_168560             12   12     12   12             12              12   
s_21005              13   13     13   13             13              13   
s_96857              13   13     13   13             13              13   
s_118668             14   14     14   14             14              14   

            price  categ  tranche de prix  taille_panier_session  achat_mens  \
session_id                                                                     
s_136877       11     11               11                     11          11   
s_137262       11     11               11                     11          11   
s_21641        11     11               11                     11          11   
s_65873        11     11               11                     11          11   
s_86536        11     11               11                     11          11   
s_93892        12     12               12                     12          12   
s_168560       12     12               12                     12          12   
s_21005        13     13               13                     13          13   
s_96857        13     13               13                     13          13   
s_118668       14     14               14                     14          14   

            achat_an  freq_mens  
session_id                       
s_136877          11         11  
s_137262          11         11  
s_21641           11         11  
s_65873           11         11  
s_86536           11         11  
s_93892           12         12  
s_168560          12         12  
s_21005           13         13  
s_96857           13         13  
s_118668          14         14

In [ ]:
nb_date_session = vente_normale.groupby('session_date').count()
nb_date_session.sort_values(by='client_id', inplace=True)
nb_date_session.shape

In [ ]:
vente_normale.shape

In [ ]:
session = vente_normale.groupby('session_id').count()
session.shape

In [ ]:
client = vente_normale.groupby(['client_id']).count()
client.shape

In [ ]:
produit = vente_normale.groupby(['id_prod']).count()
produit.shape

In [ ]:
chiffre_affaires = vente_normale['price'].sum()
chiffre_affaires

<p style='color:#88cc8d;text-align:center'>
    313 552 session_date et lignes<br/>
    157 789 sessions<br/> 
    8 596 clients acheteurs<br/> 
    3 261 références produits vendus<br/>
    Le chiffre d'affaires est de 5 362 673€<br/>
</p>

[Haut de page](#sommaire)